src/components/TodoHeader/TodoHeader.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { ChangeEvent, FormEvent, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { ErrorType } from '../../types/ErrorType';
+
+type Props = {
+  todos: Todo[],
+  isInputDisabled: boolean,
+  setHasError: React.Dispatch<React.SetStateAction<ErrorType>>,
+  onTodoAdd: (userId: number, data: Todo) => void
+};
+
+const USER_ID = 9955;
+
+export const TodoHeader: React.FC<Props> = React.memo(
+  ({
+    todos,
+    isInputDisabled,
+    setHasError,
+    onTodoAdd,
+  }) => {
+    const [todoTitle, setTodoTitle] = useState('');
+    const isAllTodosActive = todos.every(todo => todo.completed === true);
+
+    const hanlderOnSubmit = (event: FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+
+      if (!todoTitle.trim()) {
+        setHasError(ErrorType.TITLE);
+
+        return;
+      }
+
+      const newTodo: Todo = {
+        id: 0,
```

Comment:
The 'id' property of a new todo should not be hardcoded to 0. Consider generating a unique ID for each new todo.